pp3/module/Application/view/application/admin/email-sending.phtml

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+?>
+<?= $this->partial('_nav.phtml'); ?>
+<h4>E-mail sending</h4>
+<p>
+<?= $this->partial('layout/flash.phtml'); ?>
+</p>
+<?php if($this->successMessage) { ?>
+<div class="panel panel-success">
+  <div class="panel-heading">Worked!</div>
+  <div class="panel-body">
+    <?php echo $this->successMessage ?>
+  </div>
+</div>
+<?php } ?>
+<form class="" method="post" action="">
+  <div class="panel panel-default">
+    <div class="panel-heading">Limit NetBeans versions:</div>
+    <div class="panel-body">
+      <p><input type="checkbox" name="onlyVerified" id="onlyVerified" style="margin-right: 1ex" <?php if($this->onlyVerified) {echo 'checked="checked"';} ?>/><label for="onlyVerified">Only consider verified entries</label></p>
```

Comment:
Should not this input field rather be of "radio" instead of "checkbox" type? Logically thinking you can either check this option or the versions listed below this field, cannot you?

Reply:
Actually the idea is that they do combine. Consider my test portal. There are 3 plugins:

- **netbeans-builder-module**: no NetBeans version associated
- **Upload Files Button**: 11.2 and 11.3 (both verified)
- **Mozilla Rhino**: 11.3 (verified), 11.2 (unverified)

This results in this email behavior:

![image](https://user-images.githubusercontent.com/2179736/197842154-f9dacb04-06ca-4d41-a471-63673b289db2.png)

No filter: 3 plugins and 2 users (you and me) would be selected. I generated a preview email, so only one email is actually sent.

![image](https://user-images.githubusercontent.com/2179736/197842439-3443aecc-5fb8-4751-bcff-13b62d0d664e.png)

Only consider plugins associated with NetBeans 11.2 gets you 2 plugins and 2 users **netbeans-builder-module** is not selected anymore

![image](https://user-images.githubusercontent.com/2179736/197842669-94f4c9c3-b07b-4935-a289-04595b56641b.png)

Only consider plugins associated with NetBeans 11.2 and where the association is verified. This gets you 1 plugin and 1 user. The **Upload Files Button** plugin is the only plugin matching that and I'm listed as the only author.


Does that make sense to you?

Reply:
Yes, makes sense now. Sorry, I probably didn't have enough coffee back then.